Library/Homebrew/tap.rb

```diff
@@ -280,4 +305,70 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreTap < Tap
+  # @private
+  def initialize
+    @user = "Homebrew"
+    @repo = "homebrew"
+    @name = "Homebrew/homebrew"
+    @path = HOMEBREW_REPOSITORY
+  end
+
+  include Singleton
```

Comment:
> That is because you haven't convince me on such issue yet.

Sure, I just think at some point it's more productive to seek to reach a middle ground rather than arguing about it indefinitely.



> Apologize if I offend you in anyway.

That's OK, I appreciate the apology, thanks.

> Since writing singleton on my own is not too hard. I will go ahead on this approach like below

Looks good :+1: